stdlib/2and3/_typeshed/__init__.pyi

```diff
@@ -52,9 +55,15 @@ OpenBinaryModeReading = Literal[
 ]
 OpenBinaryMode = Union[OpenBinaryModeUpdating, OpenBinaryModeReading, OpenBinaryModeWriting]
 
-@type_check_only
 class HasFileno(Protocol):
     def fileno(self) -> int: ...
 
 FileDescriptor = int
 FileDescriptorLike = Union[int, HasFileno]
+
+class SupportsRead(Protocol[_T_co]):
+    def read(self, __length: int = ...) -> _T_co: ...
+class SupportsReadline(Protocol[_T_co]):
+    def readline(self, __length: int = ...) -> _T_co: ...
+class SupportsWrite(Protocol[_T_contra]):
+    def write(self, __s: _T_contra) -> int: ...
```

Comment:
While implementations hardly ever care about the return value, I included it, in case some implementation does.